src/libfsm/print/dot.c

```diff
@@ -72,20 +72,19 @@ singlestate(FILE *f, const struct fsm *fsm, fsm_state_t s)
 				fsm->opt->prefix != NULL ? fsm->opt->prefix : "",
 				e.state);
 
-			dot_escputc_html(f, fsm->opt, e.symbol);
+			dot_escputc_html(f, fsm->opt, (char) e.symbol);
 
 			fprintf(f, "> ];\n");
 		}
 
-		return;
+		return 0;
```

Comment:
Should you check `ferror` here?

Reply:
It's not intentional, you're absolutely right, I missed a bunch. Thank you!

Reply:
Wait, no. Those are static functions; ferror is checked in the caller. That's fine, isn't it?

Reply:
That makes sense.  I found it confusing that you were checking for some errors and not others, so returning `0` doesn't mean "no error" so much as "no non-fprintf error".

You explained this in the PR description, but I somehow didn't pick up on it, and reading through the code I wasn't sure why you were checking `ferror` in some places and not others, and why there were statically linked functions that returned `int` and others that returned `void`.

I think if you add a comment or two in the code to explain all of this, it will probably help avoid confusion when someone goes through the code later.